Source/SwiftLintFramework/Rules/Lint/EmptyMethodOverrideRule.swift

```diff
@@ -0,0 +1,172 @@
+import Foundation
+import SourceKittenFramework
+
+public struct EmptyMethodOverrideRule: SubstitutionCorrectableASTRule,
+        ConfigurationProviderRule, AutomaticTestableRule {
+    public var configuration = SeverityConfiguration(.warning)
+
+    public init() {}
+
+    static let nonTriggeringExamples = [
+        """
+        override func viewDidDisappear(_ animated: Bool) {
+            childViewController.viewDidDisappear(animated)
+        }
+        """,
+        """
+        override func viewDidDisappear(_ animated: Bool) {
+            super.viewDidDisappear(animated)
+            print("View controller did disappear")
+        }
+        """,
+        """
+        public override init() {
+            super.init()
+        }
+        """,
+        """
+        override func setUp() {
+            super.setUp()
+            urlString = "https://httpbin.org/basic-auth"
+        }
+        """
+    ].map(wrapInClass)
+
+    static let triggeringExamples = [
+        """
+        override func a(){/*comment*/super.a()}
+        """,
+        """
+        override func viewDidLoad() {
+            super.viewDidLoad()
+
+            // Do any additional setup after loading the view.
+        }
+        """,
+        """
+        override func didReceiveMemoryWarning() {
+            super.didReceiveMemoryWarning()
+            // Dispose of any resources that can be recreated.
+        }
+        """,
+        """
+        override func becomeFirstResponder() -> Bool {
+            return super.becomeFirstResponder()
+        }
+        """,
+        """
+        internal
+        class
+        override
+        func setUp() {
+            super.setUp()
+        }
+        """
+    ].map(wrapInClass)
+
+    static let corrections = Dictionary(
+        uniqueKeysWithValues: EmptyMethodOverrideRule.triggeringExamples.map { ($0, wrapInClass("")) }
+    )
+
+    public static let description = RuleDescription(
+        identifier: "empty_method_override",
+        name: "Empty Method Override",
+        description: "Methods that don't do anything but call `super` can be removed",
+        kind: .lint,
+        nonTriggeringExamples: EmptyMethodOverrideRule.nonTriggeringExamples,
+        triggeringExamples: EmptyMethodOverrideRule.triggeringExamples,
+        corrections: EmptyMethodOverrideRule.corrections
+    )
+
+    public func violationRanges(
+        in file: File,
+        kind: SwiftDeclarationKind,
+        dictionary: [String: SourceKitRepresentable]
+    ) -> [NSRange] {
+        let overridingKinds: [SwiftDeclarationKind] = [
+            .functionMethodInstance,
+            .functionMethodClass,
+            .functionMethodStatic
+        ]
+
+        guard
+            overridingKinds.contains(kind),
+            dictionary.enclosedSwiftAttributes.contains(.override),
+            !dictionary.enclosedSwiftAttributes.contains(.public),
```

Comment:
why is there this condition?

Reply:
Consider the following example:

```swift
class A {
    internal init() {}
}

class B: A {
    public override init() {
        super.init()
    }
}
```

In this case, the empty override is used to expose a method that has a more restrictive ACL in the superclass.

I'm not sure if we should address this, but I realized the same should apply for `open` declarations.